src/components/HeartComponent/HeartComponent.tsx

```diff
@@ -3,7 +3,7 @@ import { FaHeart, FaRegHeart } from 'react-icons/fa';
 import './HeartComponent.scss';
 
 type HeartProps = {
-  isLiked: boolean;
+  isLiked: boolean | null;
```

Comment:
I do not think HeartComponent should support null, rather value provided to it earlier should be set to either true or false (so false when it would be null)